File: leetcode-algorithms/027. Remove Element/solution.jl
function removeElement(nums::Array, val::Int)
    if length(nums) == 0
        return 0
    end
    i,t = 1,1
    while i <= length(nums)
        if val != nums[i]
            nums[t] = nums[i]
            t += 1
        end
        i+=1        
    end
    return t-1 # index start from 1
end;

l = [1,2,3,4,5]
println(removeElement(l,3))
println(l)

File: leetcode-algorithms/027. Remove Element/solution.py
from typing import List
def removeElement(nums: List[int], val: int) -> int:
    if not nums or not len(nums):
        return 0
    i,t = 0, 0
    while i < len(nums):
        if nums[i] != val:
            nums[t] = nums[i]
            t += 1
        i += 1
    return t

l = [1,2,3,4,5]
print(removeElement(l,3))
print(l)